itests/curio_test.go

```diff
@@ -1,355 +1,425 @@
-//	}
-//
-//	ctoken := fmt.Sprintf("%s:%s", string(apiToken), ma)
-//	err = os.Setenv("CURIO_API_INFO", ctoken)
-//	require.NoError(t, err)
-//
-//	capi, ccloser, err := rpc.GetCurioAPI(&cli.Context{})
-//	require.NoError(t, err)
-//
-//	scfg := storiface.LocalStorageMeta{
-//		ID:         storiface.ID(uuid.New().String()),
-//		Weight:     10,
-//		CanSeal:    true,
-//		CanStore:   true,
-//		MaxStorage: 0,
-//		Groups:     []string{},
-//		AllowTo:    []string{},
-//	}
-//
-//	err = capi.StorageInit(ctx, dir, scfg)
-//	require.NoError(t, err)
-//
-//	err = capi.StorageAddLocal(ctx, dir)
-//	require.NoError(t, err)
-//
-//	_ = logging.SetLogLevel("harmonytask", "DEBUG")
-//
-//	return capi, taskEngine.GracefullyTerminate, ccloser, finishCh
-//}
+import (
+	"context"
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"net"
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/docker/go-units"
+	"github.com/filecoin-project/curio/cmd/curio/rpc"
+	"github.com/filecoin-project/curio/cmd/curio/tasks"
+	"github.com/filecoin-project/curio/deps"
+	"github.com/filecoin-project/curio/harmony/harmonydb"
+	"github.com/filecoin-project/curio/lib/ffiselect"
+	"github.com/filecoin-project/curio/market/lmrpc"
+	"github.com/filecoin-project/curio/tasks/seal"
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-jsonrpc"
+	"github.com/filecoin-project/go-jsonrpc/auth"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/api/v1api"
+	miner2 "github.com/filecoin-project/lotus/chain/actors/builtin/miner"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/cli/spcli"
+	"github.com/filecoin-project/lotus/itests/kit"
+	"github.com/filecoin-project/lotus/node"
+	"github.com/filecoin-project/lotus/node/config"
+	"github.com/filecoin-project/lotus/storage/sealer/storiface"
+	"github.com/gbrlsnchs/jwt/v3"
+	"github.com/google/uuid"
+	logging "github.com/ipfs/go-log/v2"
+	manet "github.com/multiformats/go-multiaddr/net"
+	"github.com/stretchr/testify/require"
+	"github.com/urfave/cli/v2"
+	"golang.org/x/xerrors"
+)
+
+func TestCurioNewActor(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	full, miner, esemble := kit.EnsembleMinimal(t,
+		kit.LatestActorsAt(-1),
+		kit.MockProofs(),
+		kit.WithSectorIndexDB(),
+	)
+
+	esemble.Start()
+	blockTime := 100 * time.Millisecond
+	esemble.BeginMiningMustPost(blockTime)
+
+	addr := miner.OwnerKey.Address
+	sectorSizeInt, err := units.RAMInBytes("8MiB")
+	require.NoError(t, err)
+
+	sharedITestID := harmonydb.ITestNewID()
+	dbConfig := config.HarmonyDB{
+		Hosts:    []string{envElse("CURIO_HARMONYDB_HOSTS", "127.0.0.1")},
+		Database: "yugabyte",
+		Username: "yugabyte",
+		Password: "yugabyte",
+		Port:     "5433",
+	}
+	db, err := harmonydb.NewFromConfigWithITestID(dbConfig, sharedITestID)
+	require.NoError(t, err)
+
+	defer db.ITestDeleteAll()
```

Comment:
Would be better to pass `testing.T` (or `testing.T.Cleanup` / `func(onTestEnd func())`) into the test db constructor and register a t.Cleanup hook there.